### ISSUE TITLE
Generated variable name conversions

### ISSUE THREAD

**Issue by jaygelman** [vojtechhabarta/typescript-generator#148]
Is there any documentation about the rules that the generator uses to convert Java variable names with various mixtures of cases?  I've noticed that the following conversions are made:

abCde -> abCde (no conversion)
abCDE -> abCDE (no conversion)
aBcde -> abcde
aBCDE -> abcde
Abcde -> abcde
ABcde -> abcde

I've also noticed that for those variable names that are converted in some way, if there is an optional annotation on the variable, that optional annotation is lost during the conversion.  For example, with optional annotations
abCde -> abCde?
aBcde -> abcde  (the ? is lost)

**Comment by vojtechhabarta** [comment]
Typescript-generator works with _properties_ (not with _variables_) and it doesn't change their casing.

Which JSON library are you using? Is it `jackson2`? In this case typescript-generator gets list of properties from Jackson2. Properties can be created using fields or getters/setters or combination. So could you please compare generated interfaces with output from Jackson2 `ObjectMapper`? It should be consistent.

Or do you have different situation?

**Comment by jaygelman** [comment]
Thank you for pointing me in the right direction.  The library is jackson2, and I see that jackson2 is changing the case of the variables.  Interestingly, when it changes the case, the optional annotation is lost.  Perhaps, the generator is looking for the original variable name and not the name that is changed by jackson2.  I tried to use @JsonProperty to specify the name but that ended up creating two variables - one that was converted using jackson2's default rules and one that had the name specified in the @JsonProperty annotation.  In any case, I ended up just changing the name to something that jackson2 wouldn't convert in lieu of another solution.
